src/components/Tablet/Tablet.tsx

```diff
@@ -1,59 +1,35 @@
-import {useRef} from 'react';
 import cn from 'bem-cn-lite';
 
 import type {TTabletStateInfo} from '../../types/api/tablet';
-import type {ShowTooltipFunction} from '../../types/store/tooltip';
 import {getTabletLabel} from '../../utils/constants';
 import routes, {createHref} from '../../routes';
 
+import {WithPopupWrapper} from '../WithPopupWrapper/WithPopupWrapper';
 import {InternalLink} from '../InternalLink';
+import {TabletTooltipContent} from '../TooltipsContent';
 
 import './Tablet.scss';
 
 const b = cn('tablet');
 
 interface TabletProps {
     tablet?: TTabletStateInfo;
-    onMouseEnter?: (...args: Parameters<ShowTooltipFunction>) => void;
-    onMouseLeave?: VoidFunction;
 }
 
-export const Tablet = ({
-    tablet = {},
-    onMouseEnter = () => {},
-    onMouseLeave = () => {},
-}: TabletProps) => {
-    const ref = useRef(null);
-
-    const _onTabletMouseEnter = () => {
-        onMouseEnter(ref.current, tablet, 'tablet');
-    };
-
-    const _onTabletClick = () => {
-        const {TabletId: id} = tablet;
-
-        if (id) {
-            onMouseLeave();
-        }
-    };
-
+export const Tablet = ({tablet = {}}: TabletProps) => {
     const {TabletId: id} = tablet;
     const status = tablet.Overall?.toLowerCase();
 
     return (
-        <InternalLink
-            onClick={_onTabletClick}
-            to={id && createHref(routes.tablet, {id})}
+        <WithPopupWrapper
```

Comment:
It seems you wanted to make a HOC... and I think HOC would be better in this case.

Reply:
The problem with HOC is that there is a need to pass props from the wrapped component to the TooltipContent, which will be a param for that HOC